packages/epo-react-lib/src/form/SelectListbox/SelectListbox.tsx

```diff
@@ -0,0 +1,141 @@
+import { FunctionComponent, useLayoutEffect, useState } from "react";
+import { useUID } from "react-uid";
+import { useTranslation } from "react-i18next";
+import { onChangeCallback, Option } from "@/types/select-listbox";
+import IconComposer from "@/svg/IconComposer";
+import * as Styled from "./styles";
+import { useAccessibleDropdown, useOnOutsideClick } from "./hooks";
+
+interface SelectListboxProps {
+  value: string | string[] | null;
+  options: Option[];
+  onChangeCallback: onChangeCallback;
+  isDisabled?: boolean;
+  placeholder?: string;
+  labelledById?: string;
+  namespace?: string;
+  isMultiselect?: boolean;
+  maxWidth?: string;
+  width?: string;
+}
+
+const SelectListbox: FunctionComponent<SelectListboxProps> = ({
+  value = null,
+  options,
+  onChangeCallback,
+  isDisabled,
+  labelledById,
+  placeholder: unsafePlaceholder,
+  namespace,
+  isMultiselect = false,
+  maxWidth = "200px",
+  width,
+}) => {
+  const uid = namespace || useUID();
+  const {
+    isDropdownOpen,
+    setIsDropdownOpen,
+    activeIndex,
+    setActiveIndex,
+    select,
+    setIsFocus,
+    listRef,
+    buttonRef,
+  } = useAccessibleDropdown({
+    options,
+    value,
+    onChangeCallback,
+    isMultiselect,
+  });
+  const [selectWidth, setSelectWidth] = useState(0);
+  const { t } = useTranslation();
+
+  useLayoutEffect(() => {
+    if (listRef.current) {
+      setSelectWidth(listRef.current.clientWidth);
+    }
+  }, []);
+
+  const clickOutsideCallback = () => {
+    setIsDropdownOpen(false);
+    setIsFocus(false);
+  };
+
+  const { wrapperRef } = useOnOutsideClick(() => clickOutsideCallback());
+
+  const isChosen = (optionValue: string): boolean =>
+    value ? value.includes(optionValue) : false;
```

Comment:
a lot of `includes()` added in when multiselect was included since that will work on strings and arrays of strings. I guess there is an edge case where a selected value could be like `hotdog` and it would show the option `dog` selected.